### PR TITLE
Fixed record enhancements

### PR DESCRIPTION
**Pull request by eedrummer**
This represents a long series of code changes that refactored how the fixed record functionality works. The commit history includes the development and removal of the household features, so it may be desirable to squash it before committing to master.

Documentation has been updated in the wiki: https://github.com/synthetichealth/synthea/wiki/Fixed-Records Read down to the Enhanced Version section.

### CODE REVIEW

File: src/main/resources/modules/adverse_childhood_experiences_module.json
Comment (jawalonoski) [inline src/main/resources/modules/adverse_childhood_experiences_module.json:2]: Have we reviewed this module as a team?

File: src/main/resources/modules/diabetes/ymca_diabetes_prevention_program.json
Comment (jawalonoski) [inline src/main/resources/modules/diabetes/ymca_diabetes_prevention_program.json:5]: Have we reviewed this module as a team?

File: src/main/resources/modules/metabolic_syndrome_disease.json
Comment (jawalonoski) [inline src/main/resources/modules/metabolic_syndrome_disease.json:157]: Assuming the diabetes prevention submodule is included, do we really want this to be direct? What about conditional or distributed transition?

File: src/main/resources/modules/wic.json
Comment (jawalonoski) [inline src/main/resources/modules/wic.json:12]: Have we reviewed this module as a team?

File: src/test/resources/fixed_demographics/households_fixed_demographics_test.json
Comment (jawalonoski) [inline src/test/resources/fixed_demographics/households_fixed_demographics_test.json:3]: I can't figure out how this file is being used. What unit test is using it?

File: src/main/java/org/mitre/synthea/engine/Generator.java
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Generator.java:521]: Did you mean to commit this change?

File: src/main/java/org/mitre/synthea/engine/Generator.java
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Generator.java:539]: Are you sure we need to do this? The check criteria should handle this, especially if you properly set some of the `public boolean` attributes on this class.

File: src/main/java/org/mitre/synthea/engine/Generator.java
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Generator.java:343]: You should probably add:
```java
this.onlyAlivePatients = true;
this.onlyDeadPatients = false;
this.exporterRuntimeOptions.yearsOfHistory = 0;
```

File: src/main/java/org/mitre/synthea/engine/Generator.java
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Generator.java:688]: Use `Location.getStateName(String abbreviation)` instead.

File: src/main/java/org/mitre/synthea/engine/Generator.java
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/engine/Generator.java:848]: Use `Location.getStateName(String abbreviation)` instead.

**Review by jawalonoski — CHANGES_REQUESTED** [review]
I may add more comments.

Couple things:

- Don't use the CMSStateMapper thing, use the static Location methods instead.
- There are a couple of Generator changes I'm not sure about.
- There are some new modules that we should review as a team if we haven't already (I don't recall doing it)
- There is at least one unit test file that doesn't seem like it is being used.
- There are some class attributes that don't seem like they are being used, do we need them?
- Please add some Java doc to the new `org.mitre.synthea.identity` package.

File: src/main/resources/modules/adverse_childhood_experiences_module.json
Comment (eedrummer) [reply src/main/resources/modules/adverse_childhood_experiences_module.json:2]: This is a CODI leftover that should go into a separate PR. Same with the other modules. Apologies.

**Comment by eedrummer** [comment]
I think this is ready for a re-review. I think the core code is ready and I can try to find ways to boost coverage before it gets merged in.

**Review by jawalonoski — COMMENTED** [review]
Sorry, a couple more comments/questions on this... almost there.

File: src/main/java/org/mitre/synthea/export/CSVExporter.java
Comment (RobertScalfani) [reply src/main/java/org/mitre/synthea/export/CSVExporter.java:695]: I don't totally recall. I remember there was an issue that would rarely be triggered when split_records were enabled and this may be it. I think it had something to do with an encounter getting started in one record and then not getting completed before a new record would begin.

**Pull request merged**
